src/main/java/frc/robot/subsystems/Manipulator.java

```diff
@@ -0,0 +1,126 @@
+
+    private Manipulator()
+    {
+        _wristMotor         = new CANSparkMax(Constants.Manipulator.WRIST_MOTOR_CAN_ID, MotorType.kBrushless);
+        _wristEncoder       = new DutyCycleEncoder(Constants.Manipulator.WRIST_ENCODER_PORT);
+        _wristPID           = new PIDControl();
+        _twistMotor         = new CANSparkMax(Constants.Manipulator.TWIST_MOTOR_CAN_ID, MotorType.kBrushless);
+        _twistEncoder       = new DutyCycleEncoder(Constants.Manipulator.TWIST_ENCODER_PORT);
+        _twistPID           = new PIDControl();
+        _graspMotor         = new CANSparkMax(Constants.Manipulator.GRASP_MOTOR_CAN_ID, MotorType.kBrushless);
+        _intakeMotor        = new CANSparkMax(Constants.Manipulator.INTAKE_MOTOR_CAN_ID, MotorType.kBrushless);
+        _intakeSensor       = new DigitalInput(Constants.Manipulator.INTAKE_SENSOR_PORT);
+        _graspEncoder       = new DutyCycleEncoder(Constants.Manipulator.GRASP_ENCODER_PORT);
+        _graspPID           = new PIDControl();
+
+        _wristPID.setCoefficient(Coefficient.P, 0, 0, 0);
+        _wristPID.setCoefficient(Coefficient.I, 0, 0, 0);
+        _wristPID.setCoefficient(Coefficient.D, 0, 0, 0);
+        _wristPID.setInputRange(Constants.Manipulator.WRIST_MIN_ANGLE, Constants.Manipulator.WRIST_MAX_ANGLE);
+        _wristPID.setOutputRange(-1, 1);
+        _wristPID.setSetpointDeadband(1);
+
+        _twistPID.setCoefficient(Coefficient.P, 0, 0, 0);
+        _twistPID.setCoefficient(Coefficient.I, 0, 0, 0);
+        _twistPID.setCoefficient(Coefficient.D, 0, 0, 0);
+        _twistPID.setInputRange(Constants.Manipulator.TWIST_MIN_ROTATION, Constants.Manipulator.TWIST_MAX_ROTATION);
+        _twistPID.setOutputRange(-1, 1);
+        _twistPID.setSetpointDeadband(1);
+
+        _graspPID.setCoefficient(Coefficient.P, 0, 0, 0);
+        _graspPID.setCoefficient(Coefficient.I, 0, 0, 0);
+        _graspPID.setCoefficient(Coefficient.D, 0, 0, 0);
+        _graspPID.setInputRange(Constants.Manipulator.GRASP_MIN_POSITION, Constants.Manipulator.GRASP_MAX_POSITION);
+        _graspPID.setOutputRange(-1, 1);
+        _graspPID.setSetpointDeadband(1);
+    }
+
+    public void setWristAngle(double position)
+    {
+        _wristPID.setSetpoint(position, getWristAngle());
+    }
+
+    public double getWristAngle()
+    {
+        return _wristEncoder.get();
+    }
+
+    public void setTwistAngle(double position)
+    {
+        _twistPID.setSetpoint(position, getTwistAngle());
+    }
+
+    public double getTwistAngle()
+    {
+        return _twistEncoder.get();
+    }
+
+    public void setGraspPosition(double position)
+    {
+        _graspPID.setSetpoint(position, getGraspPosition());
+    }
+
+    public double getGraspPosition()
+    {
+        return _graspEncoder.get();
+    }
+
+    public void setIntakeSpeed(double speed)
+    {
+        _intakeMotor.set(speed);
+    }
+
+    public boolean isIntakeSensorActive()
+    {
+        return _intakeSensor.get();    
```

Comment:
The light sensors we use are closed by default and open when they sense something. What this means in code is that we need to invert the return value of the `get` function:

```suggestion
        return !_intakeSensor.get();    
```